src/ImageGallery.js

```diff
@@ -304,45 +304,55 @@ class ImageGallery extends React.Component {
     return 0;
   }
 
+  getSignedOffset = (index, currentIndex, total) => {
```

Comment:
shouldn't it be subtraction with modulus
https://en.wikipedia.org/wiki/Modular_arithmetic

i.e. mapping 
index = {0...N} , currentIndex = {0...N}, total = N
into
x = {-N/2, N/2}

Reply:
btw sometimes language impl modulus of negative number in different ways, which is not necessary the mathematical modulus, I would 

first map it into 
x={0...N}
then
x = x - int(N/2)
which will give more consistent result

unit test your signed offset